nonce.go

```diff
@@ -63,6 +62,26 @@ func (ns Nonces) Validate() error {
 	return nil
 }
 
+// Append a nonce to a nonces' list
+func (ns *Nonces) Append(n Nonce) {
+	*ns = append([]Nonce(*ns), n)
```

Comment:
Could this been written simply as
*ns = append(*ns, n)
??

Reply:
good catch, thanks!

Reply:
Fixed in d724db8